contracts/staking/validatorShare/ValidatorShare.sol

```diff
@@ -332,7 +352,21 @@ contract ValidatorShare is IValidatorShare, ERC20NonTradable, OwnableLockable, I
         return _eventsHub;
     }
 
-    function _sellVoucher(uint256 claimAmount, uint256 maximumSharesToBurn, bool legacy) private returns (uint256, uint256) {
+    function _getOrCachePolToken() private returns (IERC20Permit) {
```

Comment:
that just requires a messy interface workaround (or a manually written `staticcall`) bc token isn't available on IStakeManager (rather it's on StakeManagerStorage), regardless I went this way bc i wanted to explicitly revert in case POL token isn't set yet & not clash with matic - but this is mostly sanity and can be avoided since token.permit handles the revert